azure-functions-maven-plugin: Error because "hostJson" is null in v1.22.0

### Plugin name and version
`azure-functions-maven-plugin` -  version `1.22.0`.

Upgrading to azure-functions-maven-plugin version `1.22.0` (from `1.21.0`) gives an error in the function package:
```
[ERROR] Failed to execute goal com.microsoft.azure:azure-functions-maven-plugin:1.22.0:package (azfaas-package-functions) on project azfunfwkjava-boot: generate configuration files and prepare staging directory: Cannot invoke "com.fasterxml.jackson.databind.JsonNode.at(String)" because "hostJson" is null -> [Help `1]
```

with version `1.21.0` works correctly.

### Plugin configuration in your `pom.xml`

```
          <plugin>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>azure-functions-maven-plugin</artifactId>
            <configuration>
              <authType>${azfaasAuthType}</authType>
              <appName>${azfaasAppName}</appName>
              <resourceGroup>${azfaasResourceGroup}</resourceGroup>
              <region>${azfaasAppRegion}</region>
              <pricingTier>${azfaasPricingTier}</pricingTier>
              <appServicePlanName>${azfaasAppServicePlanName}</appServicePlanName>
              <subscriptionId>${azfaasSubscriptionId}</subscriptionId>
              <runtime>
                <javaVersion>${java.version}</javaVersion>
              </runtime>
              <appSettings>
                <property>
                  <name>FUNCTIONS_EXTENSION_VERSION</name>
                  <value>~4</value>
                </property>
                <property>
                  <name>FUNCTIONS_WORKER_RUNTIME</name>
                  <value>java</value>
                </property>
                <property>
                  <name>MAIN_CLASS</name>
                  <value>${start-class}</value>
                </property>
              </appSettings>
            </configuration>
            <executions>
              <execution>
                <id>azfaas-package-functions</id>
                <goals>
                  <goal>package</goal>
                </goals>
              </execution>
            </executions>
          </plugin>
          <plugin>
            <artifactId>maven-resources-plugin</artifactId>
            <executions>
              <execution>
                <id>azfaas-copy-resources</id>
                <phase>package</phase>
                <goals>
                  <goal>copy-resources</goal>
                </goals>
                <configuration>
                  <overwrite>true</overwrite>
                  <outputDirectory>
                    ${stagingDirectory}
                  </outputDirectory>
                  <resources>
                    <resource>
                      <directory>${project.basedir}/src/main/azure
                      </directory>
                      <includes>
                        <include>**</include>
                      </includes>
                    </resource>
                  </resources>
                </configuration>
              </execution>
            </executions>
          </plugin>
```

Note that the `host.json` and `local.setting.json` files of the final project are located in `${project.basedir}/src/main/azure` and not the root path of the project.

If I move the `host.json` and `local.setting.json` files to the root directory then it works.

### Expected behavior
The execution of the plugin is correct.

### Actual behavior
[ERROR] Failed to execute goal com.microsoft.azure:azure-functions-maven-plugin:1.22.0:package (azfaas-package-functions) on project azfunfwkjava-boot: generate configuration files and prepare staging directory: Cannot invoke "com.fasterxml.jackson.databind.JsonNode.at(String)" because "hostJson" is null -> [Help `1]

### Steps to reproduce the problem
- Place the `host.json` and `local.setting.json` files in a different path than the root of the project.
- Execute plugin


Now in the plugin execution logs you can see two new parameters `hostJson` and `localSettingsJson` with root path **that did not exist in version 1.21.0**:
```
[DEBUG] Configuring mojo 'com.microsoft.azure:azure-functions-maven-plugin:1.22.0:package' with basic configurator -->
[DEBUG]   (f) allowTelemetry = true
[DEBUG]   (f) appName = funfwkjava
[DEBUG]   (f) appSettings = {MAIN_CLASS=com.inditex.funfwkjava.Application, FUNCTIONS_EXTENSION_VERSION=~4, FUNCTIONS_WORKER_RUNTIME=java}
[DEBUG]   (f) auth = com.microsoft.azure.maven.model.MavenAuthConfiguration@30623109
[DEBUG]   (f) authType = auto
[DEBUG]   (f) buildDirectory = /home/fbc/workspaces/inditex/amiga4/azfaas/fun-azfunfwkjava/code/azfunfwkjava-boot/target
[DEBUG]   (f) disableAppInsights = false
[DEBUG]   (f) failsOnError = true
[DEBUG]   (f) finalName = azfunfwkjava-boot-0.1.0-SNAPSHOT
[DEBUG]   (f) hostJson = host.json
[DEBUG]   (f) localSettingsJson = local.settings.json
[DEBUG]   (f) outputDirectory = /home/fbc/workspaces/inditex/amiga4/azfaas/fun-azfunfwkjava/code/azfunfwkjava-boot/target/classes
[DEBUG]   (f) plugin = Component Descriptor: role: 'org.apache.maven.plugin.Mojo', implementation: 'com.microsoft.azure.maven.function.AddMojo', role hint: 'com.microsoft.azure:azure-functions-maven-plugin:1.22.0:add'
```
But the [documentation](https://github.com/microsoft/azure-maven-plugins/wiki/Azure-Functions:-Configuration-Details) does not say anything about new configuration parameters for the plugin.

--- a/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java
+++ b/azure-functions-maven-plugin/src/main/java/com/microsoft/azure/maven/function/AbstractFunctionMojo.java
@@ -22,6 +22,7 @@ import org.apache.commons.lang3.StringUtils;
 import org.apache.maven.artifact.Artifact;
 import org.apache.maven.plugins.annotations.Parameter;
 
+import javax.annotation.Nullable;
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.IOException;
@@ -244,6 +245,7 @@ public abstract class AbstractFunctionMojo extends AbstractAppServiceMojo {
         }
     }
 
+    @Nullable
     protected String getFunctionLibraryVersion() {
         final Set<Artifact> artifacts = project.getArtifacts();
         return artifacts.stream()
@@ -253,6 +255,7 @@ public abstract class AbstractFunctionMojo extends AbstractAppServiceMojo {
                 .orElse(null);
     }
 
+    @Nullable
     protected JsonNode readHostJson() {
         // todo: add configuration for host.json location
         final File hostJson = getHostJsonFile();
@@ -264,10 +267,13 @@ public abstract class AbstractFunctionMojo extends AbstractAppServiceMojo {
         }
     }
 
+    @Nullable
     protected FunctionExtensionVersion getBundleVersion() {
-        final JsonNode hostJson = readHostJson();
-        final JsonNode at = hostJson.at("/extensionBundle/version");
-        return at.isMissingNode() ? null : FunctionUtils.parseFunctionExtensionVersionFromHostJson(at.asText());
+        return Optional.ofNullable(readHostJson())
+                .map(jsonNode -> jsonNode.at("/extensionBundle/version"))
+                .filter(jsonNode -> !jsonNode.isMissingNode())
+                .map(jsonNode -> FunctionUtils.parseFunctionExtensionVersionFromHostJson(jsonNode.asText()))
+                .orElse(null);
     }
 
     @Override

